### DISCUSSION TITLE
HomeAssistant Errors using 2 coral's for detecting

### DISCUSSION THREAD

**Discussion by Engine1992** [blakeblackshear/frigate#3851] [Ask A Question]
Hi. 

I have the Frigate add-on installed on HomeAssistant. HomeAssistant relies on VM Proxmox. I have two TPU Coral coprocessors connected to the server. 

After starting Frigate, everything runs for about 30 minutes, detection is ok, no errors. After 30 minutes, it starts flooding with errors in the HomeAssistant console and in Frigate logs. Sometimes Frigate reconnects the Coral processors and everything goes back to normal for a while and then crashes again, and sometimes Frigate turns off completely. I attach screenshots from the HomeAssistant console, Proxmox console (coprocessors are detected all the time) and Frigate logs. I read that it may be a problem with the linux kernel version HomeAssistant is on - go in that direction or look elsewhere? Honestly, I start to run out of ideas what to do with it.

HomeAssistant Console:
![Błąd HA](https://user-images.githubusercontent.com/62816345/190587993-9f4670a9-d7a0-41de-8a89-3e556b2b11fd.jpg)

Proxmox "lsusb":
![lsusb z proxmox](https://user-images.githubusercontent.com/62816345/190588103-87a0ce9a-8cd9-4f7f-aae5-c1c45d1895ba.jpg)

Frigate logs:

> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f167ce80] error while decoding MB 46 103, bytestream -27
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1307980] cabac decode of qscale diff failed at 100 14
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1307980] error while decoding MB 100 14, bytestream 5797
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : rtsp://admin:*******.@192.168.1.25: corrupt decoded frame in stream 0
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1711140] left block unavailable for requested intra mode
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1711140] error while decoding MB 0 60, bytestream 23307
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1711140] left block unavailable for requested intra4x4 mode -1
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x55d0f1711140] error while decoding MB 0 32, bytestream 59529
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : rtsp://admin:********.@192.168.1.25: corrupt decoded frame in stream 0
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : av_interleaved_write_frame(): Connection reset by peer
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x55d0f1317300] Failed to update header with correct duration.
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x55d0f1317300] Failed to update header with correct filesize.
> [2022-09-16 09:13:29] ffmpeg.Dom_podworko.detect     ERROR   : Error writing trailer of rtmp://127.0.0.1/live/Dom_podworko: Connection reset by peer
> [2022-09-16 09:15:43] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:43] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:43] frigate.video                  ERROR   : Dom_podworko: ffmpeg process is not running. exiting capture thread...
> [2022-09-16 09:15:49] watchdog.Dom_podworko          ERROR   : Ffmpeg process crashed unexpectedly for Dom_podworko.
> [2022-09-16 09:15:49] watchdog.Dom_podworko          ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x561797c9c300] Timestamps are unset in a packet for stream 0. This is deprecated and will stop working in the future. Fix your code to set the timestamps properly
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561797caddc0] Increasing reorder buffer to 1
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : rtsp://admin:******.@192.168.1.25: corrupt decoded frame in stream 0
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] left block unavailable for requested intra4x4 mode -1
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] error while decoding MB 0 69, bytestream 272504
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] left block unavailable for requested intra4x4 mode -1
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] error while decoding MB 0 11, bytestream 80594
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : rtsp://admin:********.@192.168.1.25: corrupt decoded frame in stream 0
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] left block unavailable for requested intra mode
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [h264 @ 0x561798096740] error while decoding MB 0 75, bytestream 5844
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : av_interleaved_write_frame(): Connection reset by peer
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x561797c9c300] Failed to update header with correct duration.
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x561797c9c300] Failed to update header with correct filesize.
> [2022-09-16 09:15:49] ffmpeg.Dom_podworko.detect     ERROR   : Error writing trailer of rtmp://127.0.0.1/live/Dom_podworko: Connection reset by peer
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: Unable to read frames from ffmpeg process.
> [2022-09-16 09:15:53] frigate.video                  ERROR   : Dom_podworko: ffmpeg process is not running. exiting capture thread...
> [2022-09-16 09:16:00] watchdog.Dom_podworko          ERROR   : Ffmpeg process crashed unexpectedly for Dom_podworko.
> [2022-09-16 09:16:00] watchdog.Dom_podworko          ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
> [2022-09-16 09:16:00] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x559349502300] Timestamps are unset in a packet for stream 0. This is deprecated and will stop working in the future. Fix your code to set the timestamps properly
> [2022-09-16 09:16:00] ffmpeg.Dom_podworko.detect     ERROR   : av_interleaved_write_frame(): Connection reset by peer
> [2022-09-16 09:16:00] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x559349502300] Failed to update header with correct duration.
> [2022-09-16 09:16:00] ffmpeg.Dom_podworko.detect     ERROR   : [flv @ 0x559349502300] Failed to update header with correct filesize.
> [2022-09-16 09:16:00] ffmpeg.Dom_podworko.detect     ERROR   : Error writing trailer of rtmp://127.0.0.1/live/Dom_podworko: Connection reset by peer
> [2022-09-16 09:17:20] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:17:20] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:17:50] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:17:50] detector.coral2                INFO    : Starting detection process: 857
> [2022-09-16 09:17:50] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:17:53] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:20:30] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:20:30] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:21:00] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:21:00] detector.coral2                INFO    : Starting detection process: 907
> [2022-09-16 09:21:00] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:21:03] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:29:20] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:29:20] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:29:50] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:29:50] detector.coral1                INFO    : Starting detection process: 1025
> [2022-09-16 09:29:50] frigate.edgetpu                INFO    : Attempting to load TPU as usb:0
> [2022-09-16 09:29:53] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:33:40] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:33:40] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:34:10] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:34:10] detector.coral2                INFO    : Starting detection process: 1089
> [2022-09-16 09:34:10] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:34:13] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:35:40] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:35:40] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:36:10] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:36:10] detector.coral2                INFO    : Starting detection process: 1125
> [2022-09-16 09:36:10] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:36:15] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:36:50] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:36:50] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:37:20] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:37:20] detector.coral2                INFO    : Starting detection process: 1151
> [2022-09-16 09:37:20] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:37:23] frigate.edgetpu                INFO    : TPU found
> [2022-09-16 09:38:36] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:32840]
> [2022-09-16 09:42:18] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:32840]
> [2022-09-16 09:51:50] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...
> [2022-09-16 09:51:50] root                           INFO    : Waiting for detection process to exit gracefully...
> [2022-09-16 09:52:20] root                           INFO    : Detection process didnt exit. Force killing...
> [2022-09-16 09:52:20] detector.coral2                INFO    : Starting detection process: 1403
> [2022-09-16 09:52:20] frigate.edgetpu                INFO    : Attempting to load TPU as usb:1
> [2022-09-16 09:52:23] frigate.edgetpu                INFO    : TPU found

**Comment by NickM-27** [comment]
I see a lot of camera errors but also detection getting stuck, I'd start by looking at HASS issues but might be proxmox as well
